From fe20227639b2bacdb35237754c821632707d87db Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Fri, 2 Oct 2020 11:16:34 +0300
Subject: [PATCH] CMake: enable egl_viv plugins

Add needed define for egl_viv compile test and generate CMakeLists
for the egl_viv plugins.

Task-number: QTBUG-77809
Change-Id: I3b354faee36cbb85adcee895430eb5e8998108e1
---
 src/gui/configure.cmake                       |  2 ++
 .../eglfs/deviceintegration/CMakeLists.txt    |  4 +--
 .../eglfs_viv/CMakeLists.txt                  | 27 +++++++++++++++++
 .../eglfs_viv_wl/CMakeLists.txt               | 29 +++++++++++++++++++
 4 files changed, 60 insertions(+), 2 deletions(-)
 create mode 100644 src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/CMakeLists.txt
 create mode 100644 src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/CMakeLists.txt

diff --git a/src/gui/configure.cmake b/src/gui/configure.cmake
index 52da746d4d..40bf652f43 100644
--- a/src/gui/configure.cmake
+++ b/src/gui/configure.cmake
@@ -278,6 +278,8 @@ qt_config_compile_test(egl_viv
     LABEL "i.Mx6 EGL"
     LIBRARIES
         EGL::EGL
+    COMPILE_OPTIONS # special case
+        "-DEGL_API_FB=1" # special case
     CODE
 "
 #include <EGL/egl.h>
diff --git a/src/plugins/platforms/eglfs/deviceintegration/CMakeLists.txt b/src/plugins/platforms/eglfs/deviceintegration/CMakeLists.txt
index 7ddcb8993f..827bb51fd3 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/CMakeLists.txt
+++ b/src/plugins/platforms/eglfs/deviceintegration/CMakeLists.txt
@@ -22,13 +22,13 @@ if(QT_FEATURE_eglfs_mali)
     # add_subdirectory(eglfs_mali) # special case TODO
 endif()
 if(QT_FEATURE_eglfs_viv)
-    # add_subdirectory(eglfs_viv) # special case TODO
+    add_subdirectory(eglfs_viv)
 endif()
 if(QT_FEATURE_eglfs_rcar)
     # add_subdirectory(eglfs_rcar) # special case TODO
 endif()
 if(QT_FEATURE_eglfs_viv_wl)
-    # add_subdirectory(eglfs_viv_wl) # special case TODO
+    add_subdirectory(eglfs_viv_wl)
 endif()
 if(QT_FEATURE_eglfs_openwfd)
     # add_subdirectory(eglfs_openwfd) # special case TODO
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/CMakeLists.txt b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/CMakeLists.txt
new file mode 100644
index 0000000000..72952c6c5d
--- /dev/null
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/CMakeLists.txt
@@ -0,0 +1,27 @@
+# Generated from eglfs_viv.pro.
+
+#####################################################################
+## QEglFSVivIntegrationPlugin Plugin:
+#####################################################################
+
+qt_internal_add_plugin(QEglFSVivIntegrationPlugin
+    OUTPUT_NAME qeglfs-viv-integration
+    TYPE egldeviceintegrations
+    SOURCES
+        qeglfsvivintegration.cpp qeglfsvivintegration.h
+        qeglfsvivmain.cpp
+    DEFINES
+        EGL_API_FB=1
+        LINUX=1
+    INCLUDE_DIRECTORIES
+        ../../api
+    PUBLIC_LIBRARIES
+        Qt::Core
+        Qt::CorePrivate
+        Qt::EglFSDeviceIntegrationPrivate
+        Qt::Gui
+        Qt::GuiPrivate
+)
+
+#### Keys ignored in scope 1:.:.:eglfs_viv.pro:<TRUE>:
+# OTHER_FILES = "$$PWD/eglfs_viv.json"
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/CMakeLists.txt b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/CMakeLists.txt
new file mode 100644
index 0000000000..269bdde02c
--- /dev/null
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/CMakeLists.txt
@@ -0,0 +1,29 @@
+# Generated from eglfs_viv_wl.pro.
+
+#####################################################################
+## QEglFSVivWaylandIntegrationPlugin Plugin:
+#####################################################################
+
+qt_internal_add_plugin(QEglFSVivWaylandIntegrationPlugin
+    OUTPUT_NAME qeglfs-viv-wl-integration
+    TYPE egldeviceintegrations
+    SOURCES
+        qeglfsvivwlintegration.cpp qeglfsvivwlintegration.h
+        qeglfsvivwlmain.cpp
+    DEFINES
+        EGL_API_FB=1
+        LINUX=1
+    INCLUDE_DIRECTORIES
+        ../../api
+    LIBRARIES
+        Wayland::Server
+    PUBLIC_LIBRARIES
+        Qt::Core
+        Qt::CorePrivate
+        Qt::EglFSDeviceIntegrationPrivate
+        Qt::Gui
+        Qt::GuiPrivate
+)
+
+#### Keys ignored in scope 1:.:.:eglfs_viv_wl.pro:<TRUE>:
+# OTHER_FILES = "$$PWD/eglfs_viv_wl.json"
